### DISCUSSION TITLE
Configuring email backend to invite users sending via webmaster@localhost

### DISCUSSION THREAD

**Discussion by philcoyle123** [cvat-ai/cvat#8248] [Q&A]
Hi,

I have been trying to set up my the site (http://localhost:8080/) so that it can have the email backend configured to invite users.

I have tried setting this in base.py/email_settings.py/production.py/docker file (even tested updating the models.py file) but it seems to keep sending my emails from webhost@localhost which I believe is stopping my emails from being sent. After every update I rebuilt the image and used standard steps of compose down/up. Has anyone had a similar issue and was able to resolve it? I believe the DEFAULT_FROM_EMAIL  should resolve this issue but it seems like it is overwritten at some point.

This seems related to https://github.com/cvat-ai/cvat/issues/7436 and https://github.com/cvat-ai/cvat/pull/1929. It seems like users have gotten this working but I can't see the exact steps.

Below is what I imputed into each file:
Config added to base.py/email_settings.py/production.py:
EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
EMAIL_HOST = 'smtp.office365.com'
EMAIL_PORT = 587
EMAIL_HOST_USER = 'xxx.@hotmail.com'
EMAIL_HOST_PASSWORD = 'xx!'  # i also tested this with env variables
EMAIL_USE_TLS = True
EMAIL_USE_SSL = False
DEFAULT_FROM_EMAIL = 'xxx.@hotmail.com'
SERVER_EMAIL = 'xxx.@hotmail.com'

Config added to docker:
 cvat_server:
    container_name: cvat_server
    image: cvat/server:${CVAT_VERSION:-dev}
    restart: always
    depends_on:
      <<: *backend-deps
      cvat_opa:
        condition:
          service_started
    environment:
      <<: [*backend-env, *clickhouse-env]
      DJANGO_MODWSGI_EXTRA_ARGS: ''
      ALLOWED_HOSTS: '*'
      ADAPTIVE_AUTO_ANNOTATION: 'false'
      NUMPROCS: 2
      CVAT_ANALYTICS: 1
      CVAT_BASE_URL:
      ONE_RUNNING_JOB_IN_QUEUE_PER_USER:
      DJANGO_SETTINGS_MODULE: cvat.settings.email_settings  # cvat.settings.production or cvat.settings.email_settings
      EMAIL_BACKEND: 'django.core.mail.backends.smtp.EmailBackend'
      EMAIL_HOST: 'smtp.office365.com'
      EMAIL_PORT: '587'
      EMAIL_HOST_USER: 'xxx.@hotmail.com'
      EMAIL_HOST_PASSWORD: 'xx!'
      EMAIL_USE_TLS: 'True'
      EMAIL_USE_SSL: 'False'
      DEFAULT_FROM_EMAIL: 'xxx.@hotmail.com'
      SERVER_EMAIL: 'xxx.@hotmail.com'

My logs gave out From: webmaster@localhost, similar to the below: 
>>> from django.conf import settings
>>>
>>> from django.core.mail import send_mail
>>>
>>> result = send_mail(
...     'Test Subject',
...     'Test content',
...     None,  # Use None here to see what Django uses as the sender
...     ['xxxx@hotmail.com'],
...     fail_silently=False,
... )
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: Test Subject
From: webmaster@localhost
To: xxx@hotmail.com
Date: Wed, 31 Jul 2024 21:02:33 -0000
Message-ID: <172245975359.240.6597261055497301986@2315cc34253c>

Test content
-------------------------------------------------------------------------------
>>> print(f"Email sent: {result}")

Thanks in advance!

**Comment by mbspng** [comment]
@philcoyle123 I got this to work, but got stuck at the next issue that arises. Check my post, it explains what I did to get email sending working. But you will probably run into the same issue that comes after that. Post: https://github.com/cvat-ai/cvat/discussions/8458
